go/slice/slice_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2024 The Vitess Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package slice
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAll_EmptySlice(t *testing.T) {
+	result := All([]int{}, func(i int) bool { return i > 0 })
+	assert.True(t, result, "All should return true for empty slice")
+}
+
+func TestAll_AllElementsTrue(t *testing.T) {
+	result := All([]int{1, 2, 3}, func(i int) bool { return i > 0 })
+	assert.True(t, result, "All should return true when all elements are true")
+}
+
+func TestAll_SomeElementsFalse(t *testing.T) {
+	result := All([]int{1, 2, 0}, func(i int) bool { return i > 0 })
+	assert.False(t, result, "All should return false when some elements are false")
+}
+
+func TestAll_SingleElementFalse(t *testing.T) {
+	result := All([]int{0}, func(i int) bool { return i > 0 })
+	assert.False(t, result, "All should return false for a single false element")
+}
```

Comment:
I think it's better to refactor these tests into table driven tests instead for each function so it's much easier to have different inputs.

Reply:
Thank you for the suggestion. I've refactored the tests into table-driven tests as per your recommendation. Please take a look and let me know if everything aligns with the project standards. 
Appreciate your guidance.